src/main/kotlin/lotto/domain/LottoResult.kt

```diff
@@ -0,0 +1,24 @@
+package lotto.domain
+
+import java.math.BigDecimal
+import java.math.RoundingMode
+
+class LottoResult(val map: Map<LottoRank, Int>) {
+    fun calculateProfitRate(purchaseAmount: Int): Double {
+        val profitRate = getTotalProfit() / purchaseAmount.toDouble()
+        return BigDecimal(profitRate).setScale(SCALE, RoundingMode.DOWN).toDouble()
+    }
+
+    private fun getTotalProfit(): Int {
+        var total = 0
+        map.forEach { (lottoRank, count) ->
+            total += lottoRank.price * count
+        }
+        return total
+    }
```

Comment:
아래 처럼 map 확장함수를 활용해볼 수 있겠네요 :)
```suggestion
    private fun getTotalProfit(): Int {
        return map
            .map { (lottoRank, count) -> lottoRank.price * count }
            .sum()
    }
```